tests/garage/torch/algos/test_vpg.py

```diff
@@ -0,0 +1,103 @@
+import gym
+import pytest
+import torch
+
+from garage.envs.base import GarageEnv
+from garage.experiment import deterministic
+from garage.experiment import LocalRunner
+from garage.np.baselines import LinearFeatureBaseline
+from garage.torch.algos import VPG
+from garage.torch.policies import GaussianMLPPolicy
+from tests.fixtures import snapshot_config
+
+
+class TestVPG:
+
+    @classmethod
+    def setup_class(cls):
+        deterministic.set_seed(0)
+
+    def setup_method(self):
+        self._env = GarageEnv(gym.make('InvertedDoublePendulum-v2'))
+        self._runner = LocalRunner(snapshot_config)
+
+        policy = GaussianMLPPolicy(env_spec=self._env.spec,
+                                   hidden_sizes=[64, 64],
+                                   hidden_nonlinearity=torch.tanh,
+                                   output_nonlinearity=None)
+        self._params = {
+            'env_spec': self._env.spec,
+            'policy': policy,
+            'optimizer': torch.optim.Adam,
+            'baseline': LinearFeatureBaseline(env_spec=self._env.spec),
+            'max_path_length': 100,
+            'discount': 0.99,
+            'policy_lr': 1e-2
+        }
+
+    def teardown_method(self):
+        self._env.close()
+
+    def test_vpg_no_entropy(self):
+        """Test VPG with no_entropy."""
+        self._params['positive_adv'] = True
```

Comment:
the ValueError tests and performance tests can probably be handled easily with a [parameterized test](https://docs.pytest.org/en/latest/parametrize.html). i won't block your PR on this, but I will probably mention it next time.

for more complicated cases,  see `dict.update()` for a nice pattern for doing this:

```python
class Test:

   def setup_method(self):
        self.default_params = {
            'foo': 1.,
            'bar': 2.,
            'baz': 3.,
        }

    def test_case(self):
         self.default_params.update({
             'foo': 5.
        }
```

Reply:
fixed